app/app/create/page.tsx

```diff
@@ -1,81 +1,97 @@
 "use client";
 
-import { useCallback, useMemo, useState } from "react";
+import { useCallback, useEffect, useMemo, useState } from "react";
 import {
   CharacterCard,
   CharacterCardData,
 } from "@/components/create/character-card";
 import { Button } from "@/components/ui/button";
 import { useAppState } from "@/hooks/useAppState";
-import { Dialog, DialogTrigger } from "@/components/ui/dialog";
 import {
   CreateCollectionDialog,
   CreateCollectionResults,
 } from "@/components/create/create-collection-dialog";
+import { AlertDialog, AlertDialogTrigger } from "@/components/ui/alert-dialog";
 
 export default function Create() {
-  const collection = useAppState((state) => state.collection);
-  const { add, last, clear } = useAppState((state) => state.actions);
+  const { drafts, activeDraft } = useAppState((state) => state.collections);
+  const { create, update, destroy, publish } = useAppState(
+    (state) => state.actions.collection
+  );
   const [submit, setSubmit] = useState(false);
 
-  // TODO: refactor to support multiple collections
-  // TODO: reload the page for a new collection until I fix this in the data store
-  const namedCollection = useMemo(
-    () => ({
-      id: Math.random().toString(36).substring(7),
-      items: collection,
-    }),
+  useEffect(() => {
+    return () => destroy(activeDraft);
+  }, [create, destroy, activeDraft]);
+
+  const collection = useMemo(() => {
+    if (activeDraft) return drafts[activeDraft];
+    create("New Collection");
+    return { id: "", items: [] };
+  }, [activeDraft, create, drafts]);
+
+  const createEnabled = useMemo(
+    () =>
+      collection.items.length > 1 ||
+      (collection.items.length &&
+        Object.values(collection.items[0]).every((i) => i)),
     [collection]
   );
 
   const handleAdd = useCallback(() => {
-    add({ name: "", description: "", image: "" });
-  }, [add]);
+    update(collection.id, { name: "", description: "", image: "" });
+  }, [update, collection]);
 
   const handleChange = useCallback(
     (data: CharacterCardData) => {
-      last(data);
+      update(collection.id, data, collection.items.length - 1);
     },
-    [last]
+    [update, collection]
   );
 
   const handleCompleted = useCallback(
     (id: string, results: CreateCollectionResults) => {
       console.log("Collection created", id, results);
+      // TODO: Move the onCompleted logic that calls the solana program
+      //       to create-collection-dialog.tsx in the handlePublish callback
+      publish(id, results);
```

Comment:
@sxcamacho made a change to the app flow, and we will need to move the code you are working on to the create-collection-dialog.tsx file now instead of leaving it here. The new flow keeps the user in the popup dialog flow until we are done with the transaction.